src/library_dylink.js

```diff
@@ -453,9 +519,10 @@ var LibraryDylink = {
         return WebAssembly.instantiate(binary, info).then(function(result) {
           return postInstantiation(result.instance);
         });
-      }
+      } 
 
-      var instance = new WebAssembly.Instance(new WebAssembly.Module(binary), info);
+      var module = flags.compiledWasm ? binary : new WebAssembly.Module(binary);
```

Comment:
Above you do `binary instanceof WebAssembly.Module`.   Why not do that here too and avoid the new flag?